text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+        * ensure at least 90% unit test coverage for all implemented L2 constructs
+        * if the API/DevEx is rejected, make a list of necessary remediation tasks and update the module's tracking issue 
+
+## Stage 2 - Developer Preview 
+
+The goal of developer preview is to deliver a release candidate with a stable API to conduct user acceptance testing and accumulate sufficient usage to declare a module is ready for general availability. We will only make breaking changes to developer preview modules when required to address unforeseen customer use cases or issues. Since the potential for breaking changes still remains while in Developer Preview, the package name will keep the “-experimental” suffix during this stage.
+
+* **Stage 2 activities**:
+    * **Update Roadmap** - move to the CDK public roadmap “Developer Preview” column to indicate we need customer feedback on the module’s implementation
+    * **Observe** - monitor module utilization and GitHub issues filed against the experimental module
+    * **Quality Assurance** - fix all P0 bugs as they are  discovered and review all P1 bugs, making a clearly communicated fix/defer decision for each one 
+* **Exit Criteria**:
+    * **Bake time** - modules will spend at least 3 months in developer preview to accumulate sufficient usage data to gain confidence in the API’s ergonomics and customer acceptance
+    * **Usage** - module utilization in 2,000 or more stacks is considered sufficient usage. We will evaluate the language distribution of module use to determine if we need to target specific underrepresented programming language communities for feedback.
+    * **No P0 bugs** - no new P0 bugs in the past 4 weeks
+    * **Successful API/Docs review** - conduct the final API review/developer experience demo and invite relevant stakeholders
+        * review all AWS Lint exceptions and have a high bar for keeping them at GA
+        * review and sign-off module’s README by tech writer
+        * review and sign-off module's API by module owner
+        * if the API/DevEx is rejected, make a list of necessary remediation tasks and update the module's tracking issue
+
+## Stage 3 - GA
+
+The module is generally available with a backwards compatible guarantee across minor versions. 
+
+* **Stage 3 activities**:
+    * **Update Roadmap** - move to the CDK roadmap “shipped” column
+    * **Remove experimental label** - merge aws-module-experimental constructs into the main aws-module package, and deprecate the experimental package
+    * **Communicate availability** - 
+        * notifications via Gitter and Twitter
+        * comment on the GitHub tracking issue
+        * release notes call out new GA modules 
+        * write a blog post about how to use the module [optional but highly desired]
+    * **Observe** - monitor module utilization and GitHub issues filed against the stable module
+
+# Construct prioritization framework
```

Comment:
I would suggest we split these off into a separate doc. prioritization ≠ lifecycle.

We need to make construct library prioritization work mechanical and easy. Tenets are not sufficient.

Reply:
I'd argue that prioritization is the first step in the lifecycle, no?